### PR TITLE
feat(sdk-core): add lightning wallet functions

### PR DESCRIPTION
**Pull request by saravanan7mani**
Update lightning wallet.
Get lightning key chains.
Improve lightning util function readability.

### CODE REVIEW

File: modules/sdk-core/src/bitgo/lightning/lightningUtils.ts [hunk 1]
```diff
@@ -55,3 +89,89 @@ export function unwrapLightningCoinSpecific<V>(obj: { lnbtc: V } | { tlnbtc: V }
...
+
+  versionBytes.copy(data, 0, 0, 4);
+  return bs58check.encode(data);
+}
+
+/**
+ * Derives watch-only accounts from the master HD node for the given purposes and network.
+ */
+// https://github.com/lightningnetwork/lnd/blob/master/docs/remote-signing.md#required-accounts
+function deriveWatchOnlyAccounts(masterHDNode: utxolib.BIP32Interface, isMainnet: boolean): WatchOnlyAccount[] {
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/lightning/lightningUtils.ts]: nit
```suggestion
function deriveWatchOnlyAccounts(masterHDNode: utxolib.BIP32Interface, isMainnet: boolean): WatchOnlyAccount[] {
  // https://github.com/lightningnetwork/lnd/blob/master/docs/remote-signing.md#required-accounts
```

**Review by OttoAllmendinger — CHANGES_REQUESTED** [review]

File: modules/sdk-core/src/bitgo/lightning/codecs.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+  innerCodec: C
+): t.UnionC<[t.TypeC<{ lnbtc: C }>, t.TypeC<{ tlnbtc: C }>]> {
+  return t.union([t.type({ lnbtc: innerCodec }), t.type({ tlnbtc: innerCodec })]);
+}
+
+interface IPAddressBrand {
+  readonly IPAddress: unique symbol; // Ensures uniqueness across modules
+}
+
+export const IPAddressCodec = t.brand(
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/lightning/codecs.ts:15]:
```suggestion
export const IPAddress = t.brand(
```

File: modules/sdk-core/src/bitgo/lightning/codecs.ts [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+  t.string,
+  (input): input is t.Branded<string, IPAddressBrand> => isIP(input) !== 0, // Type guard that checks if the string is a valid IP
+  'IPAddress'
+);
+
+export type IPAddress = t.TypeOf<typeof IPAddressCodec>;
+
+export const KeyPurposeCodec = t.union([t.literal('userAuth'), t.literal('nodeAuth')], 'KeyPurpose');
```
Comment (OttoAllmendinger) [inline modules/sdk-core/src/bitgo/lightning/codecs.ts:23]:
```suggestion
export const KeyPurpose = t.union([t.literal('userAuth'), t.literal('nodeAuth')], 'KeyPurpose');
```

**Review by OttoAllmendinger — DISMISSED** [review]
we will strip the `Codec` suffix once we have fixed eslint

**Review by OttoAllmendinger — APPROVED** [review]

**Review by lzjluzijie — APPROVED** [review]

**Pull request merged**